test/unit/template-parts.spec.js

```diff
@@ -0,0 +1,579 @@
+/* Adapted from https://github.com/dy/template-parts - ISC - Dmitry Iv. */
+
+import { assert } from '@open-wc/testing';
+import { TemplateInstance, AttrPart, AttrPartList } from '../../src/js/utils/template-parts.js';
+
+const test = it;
+const is = assert.deepEqual;
+const any = (a, b) => assert.include(b, a);
+
+test('create: applies data to templated text nodes', () => {
+  const template = document.createElement('template');
+  const originalHTML = `{{x}}`;
+  template.innerHTML = originalHTML;
+  const instance = new TemplateInstance(template, { x: 'Hello world' });
+  is(template.innerHTML, originalHTML);
+  const root = document.createElement('div');
+  root.appendChild(instance);
+  is(root.innerHTML, `Hello world`);
+});
+
+test('create: can render into partial text nodes', () => {
+  const template = document.createElement('template');
+  const originalHTML = `Hello {{x}}!`;
+  template.innerHTML = originalHTML;
+  const instance = new TemplateInstance(template, { x: 'world' });
+  is(template.innerHTML, originalHTML);
+  const root = document.createElement('div');
+  root.appendChild(instance);
+  is(root.innerHTML, `Hello world!`);
+});
+
+test('create: can render nested text nodes', () => {
```

Comment:
suggestion(non-blocking): When I read this, it seems to be about supporting rendering HTML + templated values. Consider rewording as something like
> create: can render HTML with templated text